src/controller/eme-controller.ts

```diff
@@ -890,27 +895,66 @@ class EMEController extends Logger implements ComponentAPI {
           keyId = status;
           status = temp;
         }
-        const keyIdWithStatusChange = arrayToHex(
-          'buffer' in keyId
-            ? new Uint8Array(keyId.buffer, keyId.byteOffset, keyId.byteLength)
-            : new Uint8Array(keyId),
-        );
-
+        
+        let keyIdArray: Uint8Array = 'buffer' in keyId
+          ? new Uint8Array(keyId.buffer, keyId.byteOffset, keyId.byteLength)
+          : new Uint8Array(keyId);
+        
+        // Handle PlayReady little-endian key ID conversion for status comparison only
+        // Don't modify the original key ID from playlist parsing
+        if (mediaKeySessionContext.keySystem === KeySystems.PLAYREADY && keyIdArray.length === 16) {
+          keyIdArray = new Uint8Array(keyIdArray);
+          changeEndianness(keyIdArray);
+        }
+        
+        const keyIdWithStatusChange = arrayToHex(new Uint8Array(keyIdArray));
```

Comment:
Here too (keyIdArray is already a Uint8Array).

```suggestion
        const keyIdWithStatusChange = arrayToHex(keyIdArray);
```

Reply:
Still have an unnecessary `new Uint8Array` here.